proposals/init.md

```diff
@@ -0,0 +1,490 @@
+mutability behavior of their type. Developers who do not wish to utilize 
+`init` cannot be impacted from other types choosing to do so:
+
+```cs
+class Base
+{
+    internal readonly int Field;
+    internal int Property
+    {
+        get => Field;
+        init => Field = value; // Okay
+    }
+}
+
+class Derived : Base
+{
+    internal readonly int DerivedField;
+    internal int DerivedProperty
+    {
+        get => DerivedField;
+        init
+        {
+            DerivedField = 42; // Okay
+            Field = 13; // Error Field is readonly
+        }
+    }
+
+    public Derived()
+    {
+        Property = 42; // Okay 
+        Field = 13; // Error Field is readonly
+    }
+}
+```
+
+When `init` is used in a virtual property then all the overrides must also
+be marked as `init`. Likewise it is not possible to override a simple 
+`set` with `init`.
+
+An `interface` declaration can also particpate in `init` style initalization 
+via the following pattern:
+
+```cs
+interface IPerson
+{
+    string Name { get; init; }
+}
+
+class Init
+{
+    void M<T>() where T : IPerson, new()
+    {
+        var local = new T()
+        {
+            Name = "Jared"
+        };
+        local.Name = "Jraed"; // Error
+    }
+}
+```
+
+Restrictions of this feature:
+- The `init` accessor can only be used instance properties
+- A property cannot contain both an `init` and `set` accessor
+- All overrides of a property must have `init` if the base had `init`. This rule
+also applies to interface implementation.
+
+### Metadata encoding 
+Property `init` accessors will be emitted as a standard `set` accessor with
+the return type marked with a modreq of `IsInitOnly`. This is a new type
+which will have the following definition:
+
+```cs
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All)]
+    public sealed class InitOnlyAttribute : Attribute
+    {
+
+    }
+}
+```
+
+The compiler will match the attribute by full name. There is no requirement
+that it appear in the core library. In the case there are mulitple attributes
+by this name available then the compiler will pick the one defined in the core 
+library should one exist.
+
+## Questions
+
+### Breaking changes
+One of the main pivot points in how this feature is encoded will come down to
+the following question: 
+
+> Is it a binary breaking change to remove `init` from a `set`?
+
+Removing `init`, and thus making a field or property fully writable is never
+a source breaking change. For fields it is never a binary breaking change 
+either. Additionally removing it from a field is never a binary breaking 
+change. The only behavior up in question is whether or not this remains 
+true for a property. 
+
+If we want to make the removal of `init` from a property a compatible change
+then it will force our hand on the modreq vs. attributes decision below. If 
+one the other hand this is seen as a non-interesting scenario then this will 
```

Comment:
```suggestion
then it will force our hand on the modreq vs. attributes decision below. If, 
on the other hand, this is seen as a non-interesting scenario then this will 
```